### PR TITLE
Update ephemeral instances workflow

### PR DESCRIPTION
**Pull request by dafydd-t**
**What is this feature?**

Update the ephemeral instances workflow to use a new custom action we created in the ephemeral instances repo. 

Workflow is defined in [this repo](https://github.com/grafana/ephemeral-grafana-instances-github-action/issues/74)

Since the workflow is  defined in a private repo, we close it before running it: GitHub actions cannot directly reference a custom github action from a private repo.

This PR also consolidates the two ephemeral workflows into one: We handle PR closed events in the same workflow now.

I am also updating the CODEOWNERS for this file.

**Why do we need this feature?**

Having the custom action in one place makes it easier to update, rather than having to copy changes across multiple repos.

**Who is this feature for?**

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/ephemeral-grafana-instances-github-action/issues/74

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by tolzhabayev** [comment]
Looks like your codeowners file is outdated, sync with main?

**Review by hairyhenderson — APPROVED** [review]
:tada:

File: .github/workflows/ephemeral-instances-pr-comment.yml [hunk 1]
```diff
@@ -52,34 +48,14 @@ jobs:
           ref: main
           path: ephemeral
 
-      - name: Setup Go
-        uses: actions/setup-go@v4
+      - name: build and deploy ephemeral instance
+        uses: ./ephemeral
```
Comment (papagian) [inline .github/workflows/ephemeral-instances-pr-comment.yml:52]: out of curiosity: this refers to the path in the previous step, right?

Comment (dafydd-t) [reply .github/workflows/ephemeral-instances-pr-comment.yml:52]: yes!

**Pull request merged**